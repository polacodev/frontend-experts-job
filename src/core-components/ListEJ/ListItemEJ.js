import React from 'react';
import { TouchableOpacity } from 'react-native';
import { ListItem, Body, Right } from 'native-base';

import IconEJ from '../IconEJ/IconEJ';
import TextEJ from '../TextEJ/TextEJ';

import color from '../../config/color/color';

const ListItemEJ = ({
  user,
  rightIcon = '',
  leftIcon = '',
  onPressItem = () => ({}),
  onPressRightIcon = () => ({}),
}) => {
  const userName = user.lastName ? `${user.name} ${user.lastName}` : user.name;
  const userInfo = `${user.workarea} | ${user.cellphone}`;
  return (
    <ListItem avatar>
      <IconEJ iconName={leftIcon} size={20} color={color.icon} />
      <Body>
        <TouchableOpacity onPress={() => onPressItem(user)}>
          <TextEJ type="normal">{userName}</TextEJ>
          <TextEJ type="note">{user.email}</TextEJ>
        </TouchableOpacity>
      </Body>
      <Right>
        <IconEJ
          onPressIcon={() => onPressRightIcon(user)}
          iconName={rightIcon}
          size={20}
          color={color.icon}
        />
        <TextEJ type="note">{userInfo}</TextEJ>
      </Right>
    </ListItem>
  );
};

export default ListItemEJ;
