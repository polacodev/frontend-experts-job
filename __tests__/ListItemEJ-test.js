import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';

import ListItemEJ from '../src/core-components/ListEJ/ListItemEJ';
jest.useFakeTimers();
describe('<ListItemEJ />', () => {
  const user = {
    name: 'Mark',
    lastName: 'Stobbe',
    workarea: 'Developer',
    cellphone: '75412587',
  };
  test('renders <ListItemEJ /> correctly', () => {
    const tree = renderer
      .create(
        <ListItemEJ
          user={user}
          rightIcon="check-circle-o"
          leftIcon="times-circle-o"
          onPressItem={() => ({})}
          onPressRightIcon={() => ({})}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call onPressItem and onPressRightIcon events', async () => {
    const onPressItem = jest.fn();
    const onPressRightIcon = jest.fn();

    const { findByTestId } = render(
      <ListItemEJ
        user={user}
        rightIcon="check-circle-o"
        leftIcon="times-circle-o"
        onPressItem={onPressItem}
        onPressRightIcon={onPressRightIcon}
      />,
    );
    fireEvent.press(await findByTestId('button-body'), onPressItem);
    fireEvent.press(await findByTestId('button-icon'), onPressRightIcon);
    expect(onPressItem).toHaveBeenCalled();
    expect(onPressRightIcon).toHaveBeenCalled();
  });
});
