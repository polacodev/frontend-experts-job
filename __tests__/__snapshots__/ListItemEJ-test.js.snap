// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemEJ /> renders <ListItemEJ /> correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    avatar={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 16,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#00796b",
              "fontSize": 20,
            },
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="FontAwesome"
    >
      
    </Text>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 1,
          "marginLeft": 15,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="button-body"
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
        >
          Mark Stobbe
        </Text>
        <Text
          note={true}
          numberOfLines={1}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "paddingBottom": 10,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="button-icon"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignSelf": null,
                  "color": "#c9c8cd",
                  "fontSize": 22,
                },
                Object {
                  "color": "#00796b",
                  "fontSize": 20,
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="FontAwesome"
        >
          
        </Text>
      </View>
      <Text
        note={true}
        numberOfLines={1}
        style={
          Object {
            "alignSelf": null,
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "200",
          }
        }
        uppercase={false}
      >
        Developer | 75412587
      </Text>
    </View>
  </View>
</View>
`;
